store/store.go

```diff
@@ -1178,6 +1197,38 @@ var download = func(name, downloadURL string, user *auth.UserState, s *Store, w
 	return err
 }
 
+// downloadDeltas downloads the deltas associated with a downloadInfo, returning the paths.
+func (s *Store) downloadDeltas(name string, downloadDir string, downloadInfo *snap.DownloadInfo, pbar progress.Meter, user *auth.UserState) ([]string, error) {
+	deltaPaths := []string{}
+
+	// Initially we only download our first supported format (xdelta).
+	deltaFormat := s.deltaFormats[0]
+
+	for _, deltaInfo := range downloadInfo.Deltas {
+		if deltaInfo.Format != deltaFormat {
+			continue
+		}
+		deltaName := fmt.Sprintf("%s_%d_%d_delta.%s", name, deltaInfo.FromRevision, deltaInfo.ToRevision, deltaInfo.Format)
+
+		w, err := os.Create(path.Join(downloadDir, deltaName))
+		if err != nil {
```

Comment:
Also needs to collect after itself on errors. This function may create multiple files, and these need to be removed before returning if something bad happened.